accepted/2023/net8.0-polyfills/net8.0-polyfills.md

```diff
@@ -0,0 +1,29 @@
+# .NET 8.0 Polyfill
+
+**Owner** [Immo Landwerth](https://github.com/terrajobst)
+
+This document covers which APIs we intend to ship for older versions of .NET,
+which includes .NET Standard and .NET Framework.
+
+## Polyfills
```

Comment:
Is it worth considering CompositeFormat? We have much of the code in hand for this, it would just be a matter of packaging it appropriately.

Reply:
Also, there are a bunch of attributes that the compiler depends on which need to be implemented by hand when using the new compiler on an older runtime. Those really should be provided down-level. Here's the set we're having to maintain ourselves, and that I've seen replicated in other code bases:

d----           2/24/2023     7:26                CallerAttributes
d----           2/24/2023     7:26                DiagnosticAttributes
d----           2/24/2023     7:26                ExperimentalAttribute
d----           2/24/2023     7:26                IsExternalInit
d----           2/24/2023     7:26                SkipLocalsInitAttribute
d----           2/24/2023     7:26                StringSyntaxAttribute
d----           2/24/2023     7:26                TrimAttributes

Reply:
> Is it worth considering CompositeFormat? We have much of the code in hand for this, it would just be a matter of packaging it appropriately.

I don't think so.  It's not an exchange type, it's not an abstraction, etc.  It's new in .NET 8, integrated with overloads of existing methods like string.Format.  And its implementation is heavily based on types new in .NET 6, e.g. the interpolated string handlers, ISpanFormattable, etc.

Reply:
For non-exchange types, user-local or third-party definition won't lose their functionality. Exchange types would need compatible definition and proper forwarding logic in order to be used in libraries.

What about `DateOnly` and `TimeOnly` then? The inability to polyfill members on existing `DateTime` type?